Performance/Benchmarks/Benchmarks/GRPCSwiftBenchmark/Benchmarks.swift

```diff
@@ -14,24 +14,128 @@
  * limitations under the License.
  */
 import Benchmark
-import Foundation
+import GRPCCore
 
 let benchmarks = {
-    Benchmark.defaultConfiguration = .init(
-        metrics: [
-            .mallocCountTotal,
-            .syscalls,
-            .readSyscalls,
-            .writeSyscalls,
-            .memoryLeaked,
-            .retainCount,
-            .releaseCount,
-        ]
-    )
-    
-    // async code is currently still quite flaky in the number of retain/release it does so we don't measure them today
-    var configWithoutRetainRelease = Benchmark.defaultConfiguration
-    configWithoutRetainRelease.metrics.removeAll(where: { $0 == .retainCount || $0 == .releaseCount })
-    
-    // Add Benchmarks here
+  Benchmark.defaultConfiguration = .init(
+      metrics: [
+          .mallocCountTotal,
+          .syscalls,
+          .readSyscalls,
+          .writeSyscalls,
+          .memoryLeaked,
+          .retainCount,
+          .releaseCount,
+      ]
+  )
+  
+  // async code is currently still quite flaky in the number of retain/release it does so we don't measure them today
+  var configWithoutRetainRelease = Benchmark.defaultConfiguration
+  configWithoutRetainRelease.metrics.removeAll(where: { $0 == .retainCount || $0 == .releaseCount })
+  
+  Benchmark("Metadata - Add string") { benchmark in
+    for _ in benchmark.scaledIterations {
+      var metadata = Metadata()
+      for i in 0..<1000 {
+          metadata.addString("\(i)", forKey: "\(i)")
+      }
+    }
+  }
+  
+  Benchmark("Metadata - Remove string") { benchmark in
+    for _ in benchmark.scaledIterations {
+      var metadata = Metadata()
+      for i in 0..<1000 {
+          metadata.addString("\(i)", forKey: "\(i)")
+      }
+      
+      benchmark.startMeasurement()
+      for i in 0..<1000 {
+          metadata.removeAllValues(forKey: "\(i)")
+      }
+    }
+  }
+  
+  Benchmark("Metadata - Add binary") { benchmark in
+    let value: [UInt8] = [1, 2, 3]
+    benchmark.startMeasurement()
+    for _ in benchmark.scaledIterations {
+      var metadata = Metadata()
+      for i in 0..<1000 {
+          metadata.addBinary(value, forKey: "\(i)")
+      }
+    }
+  }
```

Comment:
Benchmark.init has an overload with a trailing setup closure which is executed once and can return some setup data: https://swiftpackageindex.com/ordo-one/package-benchmark/1.11.2/documentation/benchmark/benchmark/init(_:configuration:closure:setup:teardown:)-8kbjd